WebHostLib/tracker.py

```diff
@@ -2471,3 +2471,308 @@ def render_Starcraft2_tracker(tracker_data: TrackerData, team: int, player: int)
+
+        # add progressive items
+        for item, (prog_item, level) in non_progressive_items.items():
+            if item in inventory:
+                inventory[prog_item] = min(max(inventory[prog_item], level), progressive_item_max[prog_item])
+
+        # add progressive items
+        # Some items are progressing in order and not by the received one
+        for item, (prog_item, level) in non_progressive_items_ordering.items():
+            if item in inventory:
+                inventory[prog_item] += 1
+                
+        # add progressive charms
+        # Charms are progressing in order, not by the one which was actually got.
+        for item, (prog_item, level) in non_progressive_charms.items():
+            if item in inventory:
+                inventory[prog_item] += 1
+
+        # Completed item if we meet goal.
+        if tracker_data.get_room_client_statuses()[team, player] == ClientStatus.CLIENT_GOAL:
+            inventory["IsCompleted"] = 1
+
+    def render_HK_multiworld_tracker(tracker_data: TrackerData, enabled_trackers: List[str]):
+        inventories: Dict[Tuple[int, int], Counter[str]] = {
+            (team, player): collections.Counter({
+                tracker_data.item_id_to_name["Hollow Knight"][code]: count
+                for code, count in tracker_data.get_player_inventory_counts(team, player).items()
+            })
+            for team, players in tracker_data.get_all_players().items()
+            for player in players if tracker_data.get_slot_info(team, player).game == "Hollow Knight"
+        }
+
+        # Translate non-progression items to progression items for tracker simplicity.
+        for (team, player), inventory in inventories.items():
+            prepare_inventories(team, player, inventory, tracker_data)
+
+        options: Dict[Tuple[int, int], Dict[str, int]] = {
+            (team, player): {
+                option: value
+                for (option, value) in tracker_data.get_slot_data(team, player)["options"].items()
+            }
+            for team, players in tracker_data.get_all_players().items()
+            for player in players if tracker_data.get_slot_info(team, player).game == "Hollow Knight"
+        }
+
+        # To handle both options as one on the page
+        for (team, player), randoOptions in options.copy().items():
+            randomizeEssence = 0
+
+            for option, value in randoOptions.items():
+                if(option in ('RandomizeBossEssence', 'RandomizeWhisperingRoots')):
+                    randomizeEssence = randomizeEssence or value
+
+            options[(team, player)]["RandomizeEssence"] = randomizeEssence
+
+        return render_template(
+            "multitracker__HK.html",
+            enabled_trackers=enabled_trackers,
+            current_tracker="Hollow Knight",
+            room=tracker_data.room,
```

Comment:
```suggestion
            room=tracker_data.room,
            get_slot_info=tracker_data.get_slot_info,
```
seems that the multitracker crashes without this

Reply:
Interesting. Hadn't that issue and the value is also not used in the html part for what that segment is. The slot data are just used directly in the same method.
But yeah, let's add it. Doesn't hurt anyway.